src/applications/soundDetectionDemo/robotMovement/src/vRobotMovement.cpp

```diff
@@ -0,0 +1,369 @@
+        }
+
+        // Start the asynchronous and synchronous threads
+        yInfo() << "Starting the thread...";
+        return Thread::start();
+    }
+
+    virtual double getPeriod()
+    {
+        // Period of synchrnous thread (in seconds)
+        return update_period;
+    }
+
+    bool interruptModule()
+    {
+        // If the module is asked to stop ask the asynchrnous thread to stop
+        yInfo() << "Interrupting the module: stopping thread...";
+        return Thread::stop();
+    }
+
+    void onStop()
+    {
+        // When the asynchrnous thread is asked to stop, close ports and do
+        // other clean up
+        yInfo() << "Stopping the module...";
+        yInfo() << "Closing input port...";
+        input_port.close();
+        yInfo() << "Closing output port...";
+        output_port.close();
+        yInfo() << "Module has been closed!";
+    }
+
+    // Synchronous thread
+    virtual bool updateModule()
+    {
+        // Add any synchronous operations here, visualisation, debug out prints
+
+        // Position to send to the motor control
+        double position_to_move = 0.0;
+
+        // It is calculated according to the winner neuron and the joint step, which depends on
+        // the number of neurons of the sound source localization (180.0 / n_neurons).
+        // We add half of the step to center the position, and we substract 90.0 to move
+        // from the [0, 180] range to [-90, 90] range.
+        position_to_move = (joint_desired_position * joint_step) + half_joint_step - 90.0;
+
+        // However, we cannot move the iCub head to -90.0 degrees due to the phisical limits.
+        // Then, we check if the position_to_move exceed the joint limits
+        if(position_to_move > joint_max_limit_reduced){
+            position_to_move = joint_max_limit_reduced;
+        }
+        if(position_to_move < joint_min_limit_reduced){
+            position_to_move = joint_min_limit_reduced;
+        }
+
+        // Show the desired position and the real position
+        if(is_debug_flag == true) {
+            yDebug() << "---------------------------------------------";
+            yDebug() << "Read joint_desired_position: " << joint_desired_position;
+            yDebug() << "Position to move: " << position_to_move;
+            yDebug() << "---------------------------------------------";
+        }
+        // Send the position to the joint controller
+        joint_position_control->positionMove(joint_id, position_to_move);
+
+        // Do any other set-up required here
+
+        return Thread::isRunning();
+    }
+
+    // Asynchronous thread run forever
+    void run()
+    {
+        // YARP timestamp
+        Stamp yarpstamp;
+
+        // Forever...
+        while(true) {
+
+            const vector<AE> * q = input_port.read(yarpstamp);
+            if(!q || Thread::isStopping()) return;
+
+            // Do asynchronous processing here
+            for(auto &qi : *q) {
```

Comment:
It seems like here you are using an Address Event to send a interger value. Why do you code the integer in an event? Why not directly send the information as an INT? The problem could be that the format of the event is not what is exected - which can lead to other people using your system making errors. Can we change this to use a vector instead of an event?